### PR TITLE
fix: handle error when address format is incorrect

### LINKED ISSUE #8446: [Task]: fix handle error when address format is incorrect
### Task description

fix handle error when address format is incorrect

### PR DESCRIPTION
**Pull request by cpl121**
Closes #8446 

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/components/popups/CreateDelegationPopup.svelte [hunk 1]
```diff
@@ -99,8 +100,13 @@
             )
             transactionInfo.preparedTransactionError = undefined
         } catch (error) {
-            transactionInfo.preparedTransaction = undefined
-            transactionInfo.preparedTransactionError = error
+            if (error.message?.includes('Addresses')) {
+                addressError = error.message
+                handleError(error)
```
Comment (evavirseda) [inline packages/desktop/components/popups/CreateDelegationPopup.svelte:105]:
```suggestion
```
imo this line should be removed, as it only shows the error inline, not with a notification :pray: 
image

Comment (marc2332) [reply packages/desktop/components/popups/CreateDelegationPopup.svelte:105]: Why would you want to show a notification ? What's wrong with an inline error?

Comment (evavirseda) [reply packages/desktop/components/popups/CreateDelegationPopup.svelte:105]: i would like not to show the notification, only the inline error

Comment (marc2332) [reply packages/desktop/components/popups/CreateDelegationPopup.svelte:105]: Ah, I misunderstood.

**Review by marc2332 — CHANGES_REQUESTED** [review]
what eva said

**Review by marc2332 — CHANGES_REQUESTED** [review]
Shouldn't we also be throwing an error when the address is not correctly formatted besides not having the prefix?

image

**Review by evavirseda — APPROVED** [review]
works!

File: packages/desktop/components/popups/CreateDelegationPopup.svelte [hunk 2]
```diff
@@ -38,14 +38,15 @@
         $selectedWallet?.isTransferring
     $: validAmount = Big(rawAmount ?? 0)?.gt(0)
 
-    $: accountAddress, validAmount, prepareDelegationOutput()
+    $: accountAddress, validAmount, rawAmount, void prepareDelegationOutput()
```
Comment (begonaalvarezd) [inline packages/desktop/components/popups/CreateDelegationPopup.svelte:41]: this is tricky, if you use the slider, how many times does `rawAmount`change and how many times are you actually then preparing the output? might be a bit overkilled

**Review by marc2332 — CHANGES_REQUESTED** [review]
The error is only refreshed when there is an amount entered
image

image

File: packages/desktop/components/popups/AllotManaPopup.svelte [hunk 3]
```diff
@@ -24,28 +26,30 @@
...
         $selectedWallet?.hasConsolidatingOutputsTransactionInProgress || $selectedWallet?.isTransferring
-    $: amount, accountAddress, hasTransactionInProgress, setConfirmDisabled()
     $: asset = $visibleSelectedWalletAssets[$activeProfile?.network?.id].mana
     $: validAmount = Big(rawAmount ?? 0)?.gt(0)
-    $: accountAddress, validAmount, rawAmount, void preparedOutput()
+    $: accountAddress, validAmount, void preparedOutput()
+    $: accountAddress, void validateAddress()
 
-    $: displayManaBox = !!accountAddress && !error && validAmount
+    $: submitAllowed =
```
Comment (evavirseda) [inline packages/desktop/components/popups/AllotManaPopup.svelte:40]: this is not working... :see_no_evil: t's allowing me to send even though I don't have a correct address :pray: 
Screenshot from 2024-04-26 13-29-45

**Review by evavirseda — CHANGES_REQUESTED** [review]

**Review by evavirseda — APPROVED** [review]
:rocket:

**Review by begonaalvarezd — APPROVED** [review]

**Review by marc2332 — APPROVED** [review]

**Pull request merged**